databricks/koalas/indexing.py

```diff
@@ -1201,6 +1224,41 @@ def _select_cols_by_iterable(
             ]
             data_spark_columns = list(cols_sel)
             data_dtypes = None
+        elif all(isinstance(key, bool) for key in cols_sel) or all(
+            isinstance(key, np.bool_) for key in cols_sel
+        ):
+            if len(cast(Sized, cols_sel)) != len(self._internal.column_labels):
+                raise IndexError(
+                    "Boolean index has wrong length: %s instead of %s"
+                    % (len(cast(Sized, cols_sel)), len(self._internal.column_labels))
+                )
+            if isinstance(cols_sel, pd.Series):
+                if not cols_sel.index.sort_values().equals(self._kdf.columns.sort_values()):
+                    raise SparkPandasIndexingError(
+                        "Unalignable boolean Series provided as indexer "
+                        "(index of the boolean Series and of the indexed object do not match)"
+                    )
+                else:
+                    cols_sel = [
+                        self._internal.spark_column_for((label,))
+                        for label, col in cols_sel.iteritems()
+                        if col
+                    ]
+                    column_labels = [
+                        (self._internal.spark_frame.select(col).columns[0],) for col in cols_sel
+                    ]
+                    data_spark_columns = list(cols_sel)
+                    data_dtypes = None
```

Comment:
I'm afraid I don't think this is a right direction. Most of the metadata will be lost.

Btw, could you also try:

```py
>>> pdf = pd.DataFrame({("x", "a"): [1, 1, 3], ("x", "b"): [4, 5, 6], ("y", "c"): [7, 8, 9]})
>>> pdf
   x     y
   a  b  c
0  1  4  7
1  1  5  8
2  3  6  9
>>> pdf.loc[:, pd.Series([True, True, False], index=pd.MultiIndex.from_tuples([('y','c'),('x','a'),('x','b')]))]
   x  y
   a  c
0  1  7
1  1  8
2  3  9
```

whereas:

```py
>>> kdf.loc[:, pd.Series([True, True, False], index=pd.MultiIndex.from_tuples([('y','c'),('x','a'),('x','b')]))]
Traceback (most recent call last):
...
KeyError: "('y', 'c')"
```

Reply:
I see, then I believe https://github.com/databricks/koalas/pull/2057/files#diff-ddd6bb43744b1a2f6946e44a263f11a6b2258d69a4aaee49a39242129b6a20d1R1221-R1226 should be fixed as well?

Reply:
How do you mean to fix it?